extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -414,11 +424,32 @@ private void createAndStartPublishExecutor()
   @Override
   public TaskStatus run(final TaskToolbox toolbox) throws Exception
   {
-    // for backwards compatibility, should be remove from versions greater than 0.12.x
-    if (useLegacy) {
-      return runLegacy(toolbox);
+    try {
+      // for backwards compatibility, should be remove from versions greater than 0.12.x
+      if (useLegacy) {
+        return runInternalLegacy(toolbox);
+      } else {
+        return runInternal(toolbox);
+      }
     }
+    catch (Exception e) {
+      log.error(e, "Encountered exception while running task.");
+      Map<String, Object> context = Maps.newHashMap();
+      List<String> savedParseExceptionMessages = IndexTask.getMessagesFromSavedParseExceptions(savedParseExceptions);
+      if (savedParseExceptionMessages != null) {
+        context.put("unparseableEvents", savedParseExceptionMessages);
+      }
+      return TaskStatus.failure(
+          getId(),
+          getTaskCompletionMetrics(),
+          e.getMessage(),
```

Comment:
How about passing the full stack trace rather than a simple message? It would be much helpful to understand what's going on. If some applications need only error messages, they can parse the stack trace on their own.

Reply:
I changed this a few ways:
- The stack trace is passed in now, but TaskStatus will truncate the error message to 100 characters. This is to avoid large ZK/metadata objects.
- The full error + stack trace is provided in the TaskReport that's uploaded when the task completes